onnxruntime/core/providers/op_kernel_type_control.h

```diff
@@ -121,6 +121,29 @@ struct EnabledTypes {
   using types = boost::mp11::mp_apply<boost::mp11::mp_set_intersection, TypeSetsToConsider>;
 };
 
+/**
+* Check if the type list contains the specified type.
+*/
+template <typename TypeList, typename T>
+constexpr bool HasType() {
+  return std::is_same<boost::mp11::mp_set_contains<TypeList, T>, boost::mp11::mp_true>::value;
```

Comment:
>mp_set_contains [](start = 35, length = 15)

should probably use mp_contains, or assert or ensure that TypeList is in fact a type set.

Reply:
curious about what the compile-time performance difference is between searching a list vs. converting to a set and searching a set
but if we expect the size of the list to be small, maybe just prioritize simplicity

---
In reply to: [569702679](https://github.com/microsoft/onnxruntime/pull/6547#discussion_r569702679) [](ancestors = 569702679)

Reply:
diff would be 'contains' on a list does a count, whereas for a set it exits as soon as it finds a match. how much time that costs if for example it's the full list of element types in AllTensorTypes I don't know. given we expect the types to be created via this infrastructure and to be in a set I'll add a static_assert that it's a set for now. 

---
In reply to: [569704918](https://github.com/microsoft/onnxruntime/pull/6547#discussion_r569704918) [](ancestors = 569704918,569702679)